src/NServiceBus.Persistence.DynamoDB/Serialization/Mapper.cs

```diff
@@ -49,10 +61,28 @@ static class Mapper
         static void ThrowInvalidOperationExceptionForInvalidRoot(Type type)
             => throw new InvalidOperationException($"Unable to serialize the given type '{type}' because the json kind is not of type 'JsonValueKind.Object'.");
 
-        public static TValue? ToObject<TValue>(Dictionary<string, AttributeValue> attributeValues)
+        public static TValue? ToObject<TValue>(Dictionary<string, AttributeValue> attributeValues, JsonTypeInfo<TValue> jsonTypeInfo)
+        {
+            var jsonObject = ToNodeFromMap(attributeValues);
+            return jsonObject.Deserialize(jsonTypeInfo);
+        }
+
+        public static TValue? ToObject<TValue>(Dictionary<string, AttributeValue> attributeValues, JsonSerializerOptions? options = null)
+        {
+            var jsonObject = ToNodeFromMap(attributeValues);
+            return jsonObject.Deserialize<TValue>(options ?? DefaultOptions);
+        }
+
+        public static object? ToObject(Dictionary<string, AttributeValue> attributeValues, Type returnType, JsonSerializerOptions? options = null)
+        {
+            var jsonObject = ToNodeFromMap(attributeValues);
+            return jsonObject.Deserialize(returnType, options ?? DefaultOptions);
+        }
+
+        public static object? ToObject(Dictionary<string, AttributeValue> attributeValues, Type returnType, JsonSerializerContext context)
```

Comment:
do we need all these overloads that all accept different option types?

Reply:
Yes. Like I said before during the call these are also overloads that are available on the json serializer and have been thought through there already. No need to think this through again from our perspective